cpp/src/strings/convert/convert_integers.cu

```diff
@@ -231,15 +234,14 @@ struct integer_to_string_fn
             return;
         }
         bool is_negative = value < 0;
-        value = abs(value);
         constexpr IntegerType base = 10;
         constexpr int MAX_DIGITS = 20; // largest 64-bit integer is 20 digits
         char digits[MAX_DIGITS]; // place-holder for digit chars
         int digits_idx = 0;
-        while( value > 0 )
+        while( value != 0 )
```

Comment:
File this away for later, but it may be worth looking at how libc++ implements integer to string conversion to avoid the `abs` problem: https://github.com/llvm-mirror/libcxx/blob/ad464887b93526235ce06bbf97cd396ed6232b9a/include/charconv#L384